File: sbe/binance/spot_3_2/generated-python/spot_sbe/PercentPriceBySideFilter.py
"""Generated SBE (Simple Binary Encoding) message codec."""

import struct
from typing import List, Optional, Tuple
from io import BytesIO

class PercentPriceBySideFilter:
    """SBE message: PercentPriceBySideFilter."""

    TEMPLATE_ID = 3
    SCHEMA_ID = 3
    SCHEMA_VERSION = 2
    BLOCK_LENGTH = 37

    def __init__(self):
        self.filter_type = None
        self.multiplier_exponent = None
        self.bid_multiplier_up = None
        self.bid_multiplier_down = None
        self.ask_multiplier_up = None
        self.ask_multiplier_down = None
        self.avg_price_mins = None

    def encode(self) -> bytes:
        """Encode the message to bytes."""
        buffer = BytesIO()

        if self.multiplier_exponent is not None:
            buffer.write(struct.pack('<b', self.multiplier_exponent))
        if self.bid_multiplier_up is not None:
            buffer.write(struct.pack('<q', self.bid_multiplier_up))
        if self.bid_multiplier_down is not None:
            buffer.write(struct.pack('<q', self.bid_multiplier_down))
        if self.ask_multiplier_up is not None:
            buffer.write(struct.pack('<q', self.ask_multiplier_up))
        if self.ask_multiplier_down is not None:
            buffer.write(struct.pack('<q', self.ask_multiplier_down))
        if self.avg_price_mins is not None:
            buffer.write(struct.pack('<i', self.avg_price_mins))

        return buffer.getvalue()

    def decode(self, data: bytes) -> None:
        """Decode the message from bytes."""
        buffer = BytesIO(data)

        self.multiplier_exponent = struct.unpack('<b', buffer.read(1))[0]
        self.bid_multiplier_up = struct.unpack('<q', buffer.read(8))[0]
        self.bid_multiplier_down = struct.unpack('<q', buffer.read(8))[0]
        self.ask_multiplier_up = struct.unpack('<q', buffer.read(8))[0]
        self.ask_multiplier_down = struct.unpack('<q', buffer.read(8))[0]
        self.avg_price_mins = struct.unpack('<i', buffer.read(4))[0]

File: sbe/binance/spot_3_2/generated-php/spot_sbe/PercentPriceBySideFilter.php
<?php
/**
 * Generated SBE (Simple Binary Encoding) message codec.
 */

class PercentPriceBySideFilter
{
    public const TEMPLATE_ID = 3;
    public const SCHEMA_ID = 3;
    public const SCHEMA_VERSION = 2;
    public const BLOCK_LENGTH = 37;

    public int|float|array|null $filterType = null;
    public int|float|array|null $multiplierExponent = null;
    public int|float|array|null $bidMultiplierUp = null;
    public int|float|array|null $bidMultiplierDown = null;
    public int|float|array|null $askMultiplierUp = null;
    public int|float|array|null $askMultiplierDown = null;
    public int|float|array|null $avgPriceMins = null;

    public function encode(): string
    {
        $buffer = '';

        if ($this->multiplierExponent !== null) {
            $buffer .= pack('c', $this->multiplierExponent);
        }
        if ($this->bidMultiplierUp !== null) {
            $buffer .= pack('q', $this->bidMultiplierUp);
        }
        if ($this->bidMultiplierDown !== null) {
            $buffer .= pack('q', $this->bidMultiplierDown);
        }
        if ($this->askMultiplierUp !== null) {
            $buffer .= pack('q', $this->askMultiplierUp);
        }
        if ($this->askMultiplierDown !== null) {
            $buffer .= pack('q', $this->askMultiplierDown);
        }
        if ($this->avgPriceMins !== null) {
            $buffer .= pack('l', $this->avgPriceMins);
        }

        return $buffer;
    }

    public function decode(string $data): void
    {
        $offset = 0;

        $this->multiplierExponent = unpack('c', substr($data, $offset, 1))[1];
        $offset += 1;
        $this->bidMultiplierUp = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->bidMultiplierDown = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->askMultiplierUp = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->askMultiplierDown = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->avgPriceMins = unpack('l', substr($data, $offset, 4))[1];
        $offset += 4;
    }
}
